core/src/impl/Kokkos_ViewCtor.hpp

```diff
@@ -273,6 +260,32 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
   }
 };
 
+template <typename... P>
+auto add_properties(const ViewCtorProp<P...> &view_ctor_prop) {
+  return view_ctor_prop;
+}
+
+template <typename... P, typename Property, typename... Properties>
+auto add_properties(const ViewCtorProp<P...> &view_ctor_prop,
+                    const Property &property,
+                    const Properties &... properties) {
```

Comment:
Why should we provide a variadic function?
Is adding a single property not sufficient?

Reply:
The design is motivated by the code to be replaced. We were previously writing stuff like
```
using alloc_prop = Kokkos::Impl::ViewCtorProp<
        P...,
        std::conditional_t<alloc_prop_input::has_label,
                           std::integral_constant<unsigned, 0>, std::string>,
        std::conditional_t<alloc_prop_input::has_memory_space,
                           std::integral_constant<unsigned, 1>,
                           typename traits::device_type::memory_space>,
        std::conditional_t<alloc_prop_input::has_execution_space,
                           std::integral_constant<unsigned, 2>,
                           typename traits::device_type::execution_space>>;
```
and it seems sensible to me to continue supporting to add all desired properties in one go. Adding a single property would be sufficient but a variadic function is convenient and allows for more concise code.